src/components/reservation/reservation-modal.tsx

```diff
@@ -107,67 +123,73 @@ const ReservationModal = ({ manifestation }: ReservationModalProps) => {
       )}
       closeModalAriaLabelText={t("reservationModalCloseModalAriaLabelText")}
     >
-      {reservationDidSuccess && patron && (
+      {reservationResult === "success" && reservationDetails && patron && (
         <ReservationSucces
           modalId={reservationModalId(faustId)}
           title={titles.main[0]}
           preferredPickupBranch={getPreferredLocation(
             patron.preferredPickupBranch,
             branchData
           )}
+          numberInQueue={String(reservationDetails.numberInQueue)}
```

Comment:
Please use the original types from the response if possible.

As I see it is typed as [an optional number from the API spec](https://github.com/reload/dpl-react/blob/f46afa9f427e45f37afe3725e52322e4c849a5d2/src/core/fbs/model/reservationDetailsV2.ts#L16-L16). I do not see a reason to change this.